### PR TITLE
fix boss_unit death tracking

### PR DESCRIPTION
**Pull request by XVhc6A**
### Brief description of the changes:
Before this change, boss unit deaths aren't properly tracked. This change fixes that issue.

I tested this locally, no issues.

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

File: functions/boss_unit.lua [hunk 1]
```diff
@@ -35,8 +35,9 @@ local function on_entity_damaged(event)
 	entity.health = entity.health + event.final_damage_amount
 	boss.health = boss.health - event.final_damage_amount
 	if boss.health <= 0 then
-		global.boss_units[entity.unit_number] = nil
-		entity.die()
+		local unit_number = entity.unit_number
+		entity.die(event.cause.force, event.cause)
```
Comment (developer-8) [inline functions/boss_unit.lua:39]: `event.cause` can be `nil` and will raise an error in that case

Comment (XVhc6A) [reply functions/boss_unit.lua]: I've altered this w/ an if-statement, but what are some scenarios where the cause is nil? just if we do it by code?

Comment (developer-8) [reply functions/boss_unit.lua]: for example, a cluster grenade

Comment (developer-8) [reply functions/boss_unit.lua]: I think your fix will not work, if `event.cause = nil`, the `event.cause.force` will cause error

Comment (XVhc6A) [reply functions/boss_unit.lua]: how did you know that cluster grenades do that?

Sorry, the previous commit is not what I intended. I fixed it. just fallback to event.die() now

Comment (developer-8) [reply functions/boss_unit.lua]: Recently I was doing tests for a special game and found it

**Comment by developer-8** [comment]
I tested this code
```
		global.boss_units[entity.unit_number] = nil
		if event.cause then
			entity.die(event.cause.force, event.cause)
		else
			entity.die()
		end
```
it count boss kill to killscore 
so 
> You need to grab their unit_number before you die them.

has nothing to do with it.

**Comment by XVhc6A** [comment]
I tested that, and confirmed it is working.  much cleaner, thank you for your feedback.

**Review by amannm — APPROVED** [review]

**Review by clifffrey — APPROVED** [review]

**Pull request merged**